scripts/firmware.py

```diff
@@ -0,0 +1,155 @@
+        return f"{color}{text}{self.RESET}"
+
+
+class Menu:
+    def __init__(self, title: str, menu_items: dict):
+        self.color = Color()
+        self.title = title
+        self.menu_items = menu_items
+
+    def display(self):
+        while True:
+            # Print menu header
+            print("=" * 40)
+            print(self.color.colored_text(self.title, self.color.RED))
+            print("=" * 40)
+
+            # Print menu items
+            for key, (description, _) in self.menu_items.items():
+                print(f"{key}. {description}")
+            print("0. Exit")
+            print("=" * 40)
+
+            # Get user input
+            try:
+                choice = int(input("Select an option: ").strip())
+            except ValueError:
+                print(f"{Color.RED}Invalid input. Please enter a number.{Color.RESET}")
+                continue
+
+            # Handle exit
+            if choice == 0:
+                print("Exiting...")
+                break
+
+            # Call the corresponding function
+            if choice in self.menu_items:
+                _, function = self.menu_items[choice]
+                print(
+                    f"{Color.GREEN}You selected: {self.menu_items[choice][0]}{Color.RESET}\n"
+                )
+                function()  # Call the associated function
+            else:
+                print(f"{Color.RED}Invalid choice. Please try again.{Color.RESET}")
+
+
+class Firmware:
+    def __init__(self) -> None:
+        self.color = Color()
+
+    def error_msg(self, message: str) -> None:
+        if message is not None and DEBUG:
+            print(self.color.colored_text("Error:", self.color.RED), message)
+        return
+
+    def success_msg(self, message: str) -> None:
+        if message is not None and DEBUG:
+            print(self.color.colored_text("Success:", self.color.GREEN), message)
+        return
+
+    def page(self, title: str) -> None:
+        if title is not None and DEBUG:
+            print(self.color.colored_text("Step:", self.color.MAGENTA), title)
+        return
+
+    def step_title(self, title: str) -> None:
+        if title is not None and DEBUG:
+            print(self.color.colored_text("Step:", self.color.YELLOW), title)
+        return
+
+    # Header for menus etc
+    def header(self):
+        print("=" * 100)
+        print(
+            " " * 35
+            + self.color.colored_text("CARTOGRAPHER FIRMWARE FLASHER", self.color.CYAN)
+        )
+        print("=" * 100)
+
+    # find can uuid from klippy.log
+    def find_can_uuid(self) -> None:
+        self.step_title("Finding CAN Device UUID")
+
+    # Check status of UUID
+    def get_can_status(self, uuid: str) -> None:
+        self.step_title("Checking CAN Device Status")
+        if not os.path.exists(KLIPPER_DIR):
+            return self.error_msg(f"Directory '{KLIPPER_DIR}' does not exist")
+        if uuid is None:
+            return

```

Comment:
This doesn't make sense, but guess it's a work in progress?
Can uuid be `None`? Not according to the signature 😄

Reply:
haha def work in progress :D